rustdoc.el

```diff
@@ -76,103 +96,68 @@ This is useful if you want to search for the name of a struct, enum or trait."
-                             ".org")))
-    (make-directory
-     (file-name-directory outputfile)
-     t)
-    (call-process "pandoc"
-                  nil
-                  "*pandoc-log*"
-                  nil
-                  (file-truename file)
-                  "--lua-filter"
-                  (file-truename rustdoc-lua-filter)
-                  "-o"
-                  (file-truename outputfile))
-    (rustdoc-remove-whitespace outputfile)))
+(if (< emacs-major-version 27)
+    (defun rustdoc--xdg-data-home ()
+      (or (getenv "XDG_DATA_HOME")
+          (concat (file-name-as-directory (getenv "HOME"))
+                  ".local/share")))
+  (fset 'rustdoc--xdg-data-home 'xdg-data-home))
 
 
 ;;;###autoload
-(defun rustdoc-remove-whitespace (outputfile)
-  "Remove blank lines from OUTPUTFILE, unless the line after is a header."
-  (condition-case nil
-      (with-temp-file outputfile
-        (insert-file-contents outputfile)
-        (goto-char (point-min))
-        (while (not (eobp))
-          (when (and (rustdoc--current-line-empty-p)
-                     (not (rustdoc--next-line-is-header-p)))
-            (kill-whole-line))
-          (forward-line)))
-    (error (message "Missing %s " outputfile))))
+(defun rustdoc-convert-current-package ()
+  "Convert the documentation for a project, and its dependencies."
+  (interactive)
+  (let* ((proj (lsp-workspace-root))
+         (docs-src (concat (file-name-as-directory proj) "target/doc"))
+         ;; FIXME: Currently, the converted files are stored inside the project.
+         ;;        I want to store them elsewhere, as many projects could share
+         ;;        the same docs. *However* that would have to be versioned, so
+         ;;        we'll have to figure out a way to coerce `<crate>-<version>`
+         ;;        strings out of cargo, or just parse the Cargo.toml file, but
+         ;;        then we'd have to review different parsing solutions.
+         ;;
+         ;;        For now, I think this is an ok solution.
+         (docs-dst (concat (file-name-as-directory proj)
+                           rustdoc-local-directory))
+         (finish-func (lambda (p)
+                        (message (format "Finished converting docs for: %s" proj)))))
+    (let ((link-tgt (concat (file-name-as-directory (rustdoc--xdg-data-home))
+                            "emacs/rust-doc/std"))
+          (link-name (concat (file-name-as-directory proj)
+                             (file-name-as-directory rustdoc-local-directory)
+                             "std")))
+      (make-symbolic-link link-tgt link-name t))
+    (async-start-process
+     "*rust doc to org*"
+     rustdoc-convert-prog
+     finish-func
+     docs-src
+     docs-dst)))
 
 
 ;;;###autoload
-(defun rustdoc-convert-current-package ()
-  "Generate and convert the documentation for the current rust package."
+(defun rustdoc-setup ()
```

Comment:
Running this returns "emacs: /home/sam/.local/bin/rustdoc-to-org-convert.sh: Exec format error"

Reply:
Oh, did you try temporarily changing `rustdoc-source-user` to `snyball` then running the `setup` function to download the resources?

Reply:
What does `/home/sam/.local/bin/rustdoc-to-org-convert.sh` look like?

Reply:
That did it, thank you!